### PR TITLE
FORMS-674: Search

### PR DESCRIPTION
**Pull request by rimi-itk**
https://jira.itkdev.dk/browse/FORMS-674

- FORMS-674: Removed use of deprecated function utf8_encode
- FORMS-674: Cleaned up and added search command

### CODE REVIEW

File: docs/SF1500.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# SF1500: Organisation
+
+<https://digitaliseringskataloget.dk/integration/sf1500>
+
+## Example: Find Organisation by CVR number
+
+```php
+
```
Comment (jekuaitk) [inline docs/SF1500.md:8]: Should we add a working example here?

File: README.md [hunk 2]
```diff
@@ -245,6 +245,12 @@ docker run --volume ${PWD}:/app --workdir /app node:18 yarn install
 docker run --volume ${PWD}:/app --workdir /app node:18 yarn coding-standards-check
 ```
 
+### Code analysis
+
+```sh
+docker run --interactive --tty --rm --volume ${PWD}:/app --env COMPOSER_MEMORY_LIMIT=-1 itkdev/php7.4-fpm:latest composer code-analysis
```
Comment (jekuaitk) [inline README.md]: We should probably use `itkdev/php8.1-fpm:latest` here.

File: src/Service/SF1500/BrugerService.php [hunk 3]
```diff
@@ -13,108 +13,157 @@
...
+    {
+        $id = $oejebliksbillede->getObjektType()->getUUIDIdentifikator();
+        $model = new Bruger(['id' => $id]);
+        foreach ($oejebliksbillede->getRegistrering() as $registrering) {
+            foreach ($registrering->getAttributListe()->getEgenskab() as $egenskab) {
+                $model->brugernavn = $egenskab->getBrugerNavn();
+                $model->brugertype = $egenskab->getBrugerTypeTekst();
+            }
+            foreach ($registrering->getRelationListe()->getAdresser() as $adresse) {
+                // TODO
```
Comment (jekuaitk) [inline src/Service/SF1500/BrugerService.php]: Is there still more todo here?

File: src/Service/SF1500/BrugerService.php [hunk 4]
```diff
@@ -13,108 +13,157 @@
...
+            $relationListe->addToAdresser(
+                (new AdresseFlerRelationType(
+                    // Email_bruger
+                    (new UuidLabelInputType('5d13e891-162a-456b-abf2-fd9b864df96d')),
+                    // Bruger.Adresse.adresse
+                    (new UuidLabelInputType('71a08d28-3af7-4bb4-9964-bc2b76b93d64')),
+                    '4'
+                ))
+                //                    ->setReferenceID((new UnikIdType())
+                //                        ->setUUIDIdentifikator('5c5245a7-b719-439b-a25d-f35f7687d150')
```
Comment (jekuaitk) [inline src/Service/SF1500/BrugerService.php]: This should probably be removed.

**Review by jekuaitk — CHANGES_REQUESTED** [review]

File: docs/SF1500.md (hunk 1 above)
Comment (rimi-itk) [reply docs/SF1500.md:8]: Yes … when we have a working example …

File: src/Service/SF1500/BrugerService.php (hunk 3 above)
Comment (rimi-itk) [reply src/Service/SF1500/BrugerService.php]: I don't know yet. I don't know if the models will be useful.

**Review by jekuaitk — APPROVED** [review]

File: src/Service/SF1500/BrugerService.php [hunk 5]
```diff
@@ -49,80 +56,153 @@ public function list(array $ids, array $fields = []): array
...
+        return $response;
+    }
+
+    private function getManagerIds()
+    {
+        $service = $this->getService(OrganisationFunktionService::class);
+        /** @var OrganisationFunktion[] $organisationFunktioner */
+        $organisationFunktioner = $service->soeg([
+            OrganisationFunktionService::FILTER_FUNKTIONSTYPEID => $this->options['organisation-funktion-manager-id']
+        ]);
```
Comment (jekuaitk) [inline src/Service/SF1500/BrugerService.php]: Could we end up in a scenario where this list is not exhaustive?

File: src/Service/SF1500/BrugerService.php [hunk 6]
```diff
@@ -49,80 +56,153 @@ public function list(array $ids, array $fields = []): array
...
+    protected function buildModel($oejebliksbillede): Bruger
+    {
+        assert($oejebliksbillede instanceof FiltreretOejebliksbilledeType);
+        $id = $oejebliksbillede->getObjektType()->getUUIDIdentifikator();
+        $model = new Bruger(['id' => $id]);
+        foreach ($oejebliksbillede->getRegistrering() as $registrering) {
+            foreach ($registrering->getAttributListe()->getEgenskab() as $egenskab) {
+                $model->brugernavn = $egenskab->getBrugerNavn();
+                $model->brugertype = $egenskab->getBrugerTypeTekst();
+            }
```
Comment (jekuaitk) [inline src/Service/SF1500/BrugerService.php]: Do we unintentionally overwrite `brugernavn` and `brugertype` here?

File: src/Service/SF1500/BrugerService.php [hunk 7]
```diff
@@ -49,80 +56,153 @@ public function list(array $ids, array $fields = []): array
...
+            foreach (($registrering->getRelationListe()->getAdresser() ?? []) as $adresse) {
+                $model->setRelation(
```
Comment (jekuaitk) [inline src/Service/SF1500/BrugerService.php]: `setRelation` to me makes it seem it is overwritten for each `$adresse` when in fact it adds to a list of relations. Should we rename to `addRelation`?

File: src/Service/SF1500/BrugerService.php (hunk 5 above)
Comment (rimi-itk) [reply src/Service/SF1500/BrugerService.php]: I'm not sure I understand this question. We want to search for a specific type (id) and therefore search, i.e. filter, only on `OrganisationFunktionService::FILTER_FUNKTIONSTYPEID`.

File: src/Service/SF1500/BrugerService.php (hunk 6 above)
Comment (rimi-itk) [reply src/Service/SF1500/BrugerService.php]: Not unintentionally, but we assume that we have only a single item in the attribute list. I'll add a clarifying comment.

File: src/Service/SF1500/BrugerService.php (hunk 7 above)
Comment (rimi-itk) [reply src/Service/SF1500/BrugerService.php]: I'll add a clarifying comment.

**Review by jekuaitk — APPROVED** [review]

**Pull request merged**